WordPress/gutenberg
# Post Editor: Prevent popover from being hidden by metabox

Fixes #62998
Alternative to #63809 and #63834

## What?

This PR fixes an issue where popovers rendered from editor content is hidden behind meta boxes if the metabox is enabled.

## Why?

#62681 fixed the block toolbar hiding the header shadow by applying isolation: isolate; to the editor area and making the editor area a stacking context, but this meant that if the editor rendered any popovers, they would be hidden under the metabox.


## How?
This PR changes the elements that are considered as stacking contexts as follows:

Post Editor:

```html
<div class="editor-editor-interface edit-post-layout is-mode-visual has-metaboxes interface-interface-skeleton has-footer">
	<div class="interface-interface-skeleton__editor">
		<div class="interface-navigable-region interface-interface-skeleton__header"></div>
		<div class="interface-interface-skeleton__body"> 
			<div class="interface-navigable-region interface-interface-skeleton__content">
				<div class="editor-visual-editor edit-post-visual-editor"></div>
				<div class="edit-post-layout__metaboxes"></div>
			</div>
		</div>
	</div>
</div>
```

Site Editor: 

```html
<div class="editor-editor-interface edit-site-editor__editor-interface interface-interface-skeleton has-footer">
	<div class="interface-interface-skeleton__editor">
		<div class="interface-navigable-region interface-interface-skeleton__header"></div>
		<div class="interface-interface-skeleton__body"> 
			<div class="interface-navigable-region interface-interface-skeleton__content">
				<div class="editor-visual-editor edit-post-visual-editor is-iframed"></div>
			</div>
		</div>
	</div>
</div>
```

This ensures that the editor area and the media box belong to the same context, and the popover will not be obscured by either area.

## Testing Instructions

First, add the following code to `gutenberg.php`. This code adds a button to the meta box. When you click the button, it will display absolutely positioned content that intentionally jumps out of the meta box area.

```php
function my_meta_box(){
	add_meta_box( 'my_meta_box', 'My Metabox', 'my_meta_box_content', 'post' );
}
function my_meta_box_content(){
	echo '<div style="position:relative;height:500px">
		<div id="my_popover" style="display:none;border:1px solid;background:#eee;position:absolute;width:150px;height:150px;bottom:100%;left:0;">Popover!</div>
		<button id="popover_button">Popover Toggle Button</button>
	</div>
	<script>
		document.getElementById("popover_button").addEventListener("click", function() {
			const popover = document.getElementById("my_popover");
			popover.style.display = popover.style.display === "none" ? "block" : "none";
		});
	</script>
	';
}
add_action( 'admin_menu', 'my_meta_box' );
```

- Post Editor:
  - Trigger the highlight popover on the bottom-most text in your content.
  - The popover should appear above the meta box when positioned at the bottom.
  - Click a button inside the meta box. The popover should appear above the editor area.
  - Activate the meta box in the Yoast plugin and highlight the text at the very bottom of your content.
  - Verify that the issue fixed in [#62681](https://github.com/WordPress/gutenberg/pull/62681) hasn't reoccurred.
- Site Editor:
  - Verify that the issue fixed in [#62681](https://github.com/WordPress/gutenberg/pull/62681) hasn't reoccurred.

## Screenshots or screencast 


https://github.com/user-attachments/assets/71642e74-c698-47c7-8b69-20598c19a4da



> [!NOTE]
> When you merge this PR, don't forget to add the following people as co-authors, who helped resolve this issue:
> @Rishit30G @dhananjaykuber
